### DISCUSSION TITLE
How to access variables defined in top-level statement in the namespace area

### DISCUSSION THREAD

**Discussion by peymanmajidi** [dotnet/csharplang#4833] [Q&A] [answered]
```
using System;
int x = 10; // Top-level statement area in c# v9+
namespace ConsoleApp1
{

    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine(x);  // <---- how to access x 
        }
    }
}


```

How to access variables defined in the top-level statement in the namespace area?

**Comment by En3Tho** [comment, marked as answer]
Just call Console.WriteLine without any of the program/main boilerplate. Basically top level means you're already in "Main" method, you don't need to create entry point manually

**Reply by peymanmajidi** [reply]
thanks, I got the idea... but sometimes in the future if I want to access methods or vars defined in the top-level area is there any way to call them??

**Reply by bernd5** [reply]
You simply can't.
You can imagine top Level statements as regular statements in a generated class with a generated Main method.
Top Level methods are just "local functions" in the generated Main method.

Your code would be something like:

```cs
using System;

class CompilerGeneratedTopLevelClass
{
  public static void Main(){
     int x = 10; // Top-level statement area in c# v9+
  }
}

namespace ConsoleApp1
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine(x);  // <---- how to access x 
        }
    }
}
```

**Reply by peymanmajidi** [reply]
Thank you so much dear @brend5
Comprehensive answer
